packages/core/integration-tests/test/react-refresh.js

```diff
@@ -0,0 +1,144 @@
+import assert from 'assert';
+import path from 'path';
+import {
+  bundler,
+  inputFS as fs,
+  getNextBuild,
+  defaultConfig,
+  sleep
+} from '@parcel/test-utils';
+import getPort from 'get-port';
+import JSDom from 'jsdom';
+import nullthrows from 'nullthrows';
+
+const inputDir = path.join(__dirname, '/input');
+
+describe('react-refresh', function() {
+  let b,
+    root,
+    window,
+    subscription,
+    randoms = {};
+
+  before(async () => {
```

Comment:
I'd recommend making this a `beforeEach` so we don't reuse state between test runs.